sktime/forecasting/base/adapters/_neuralforecast.py

```diff
@@ -189,15 +190,37 @@ def _fit(
         if not fh.is_all_out_of_sample(cutoff=self.cutoff):
             raise NotImplementedError("in-sample prediction is currently not supported")
 
-        if self.freq == "auto" and fh.freq is None:
-            # when freq cannot be interpreted from ForecastingHorizon
-            raise ValueError(
-                f"Error in {self.__class__.__name__}, "
-                f"could not interpret freq, "
-                f"try passing freq in model initialization"
-            )
-
-        self._freq = fh.freq if self.freq == "auto" else self.freq
+        # A. freq is not auto {use this}
+        # B. freq is auto
+        #     B1. freq is infered {use this}
+        #     B2. freq is not infered
+        #         B2.1. y is date-like {use "D"}
+        #         B2.2. y is int-like  {use 1}
+        #         B2.3. other indexes; this part of code is never reached
+
+        if self.freq != "auto":
+            # A
+            self._freq = self.freq
+        else:
+            # B
+            if fh.freq:
+                # B1
+                self._freq = fh.freq
+            else:
+                # B2
+                if isinstance(y.index, (pandas.PeriodIndex, pandas.DatetimeIndex)):
+                    # B.2.1
+                    self._freq = "D"
```

Comment:
I'm afraid, but this is very questionable. Can you please explain why you suggest this?

(I would have raised an error, B2.3 case I think.)

Reply:
I would use the same argument here https://github.com/sktime/sktime/pull/6237#discussion_r1545719506 for date-like index

Reply:
I am not very sure about this actually. Suppose the original dates are weeks or months, and for some reason pandas.infer_freq fails and we pass D, will it still work correctly? The dates generated with D and the dates passed in X will not match, does that not cause failures?

Did you verify this? This seems related.

https://github.com/sktime/sktime/pull/5962#discussion_r1494942168

Reply:
I tried that in code but I've always found the input and output frequencies to align regardless of the value in `freq`

Reply:
but you solution also sounds to produce a more safer interface and I just checked, there would also be no test failures, so do you suggest I bring on these changes?

> (I would have raised an error, B2.3 case I think.)

Reply:
That was my suggestion, yes. But since you verified it to work, I am now confused because it definitely failed for me before. So, you are saying the linked code snippet is not failing for you?

Can you please share the code you tried for this case and on which package version (sktime/pandas/neuralforecast)? I'll try to replicate it on my system.

Reply:
the code snippet above does not work unless we give the model a freq of "A-DEC", however neuralforecast implemented in sktime works independent of freq 
```py
y, X = load_longley()
y_train, y_test, X_train, X_test = temporal_train_test_split(y, X, test_size=4)

model = NeuralForecastRNN(freq="D", max_steps=1, trainer_kwargs={"logger": False})

model.fit(y=y_train, X=X_train, fh=[1, 2, 3, 4])
model.predict(X=X_train)
```